lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -0,0 +1,396 @@
+    output->setEncoding(xercesc::XMLString::transcode("UTF-8"));
+    serializer->setNewLine(xercesc::XMLString::transcode("\n"));
+    //   Config would configure serialized XML data
+    config = serializer->getDomConfig();
+    config->setParameter(xercesc::XMLUni::fgDOMWRTFormatPrettyPrint, true);
+    config->setParameter(xercesc::XMLUni::fgDOMXMLDeclaration, true);
+    target = new xercesc::LocalFileFormatTarget(xercesc::XMLString::transcode(xml_file.c_str()));
+    //   Parser would validate the serialized XML against the defined XSD, with some configured parameters
+    parser = new xercesc::XercesDOMParser();
+    std::string xsd_file_path = "/home/jesus/Desktop/xsdSchema.xsd"; // TODO update this
+    std::string namespace_schema_location = eprosima::qosprof::tag::EPROSIMA_URL;
+    namespace_schema_location += " " + xsd_file_path;
+    parser->loadGrammar(xsd_file_path.c_str(), xercesc::Grammar::SchemaGrammarType, true);
+    parser->setExternalSchemaLocation(namespace_schema_location.c_str());
+    parser->setValidationScheme(xercesc::XercesDOMParser::Val_Always);
+    parser->setDoNamespaces(true);
+    parser->setDoSchema(true);
+    parser->setValidationConstraintFatal(true);
+    //   Error handler would receive the exception from the parser, and report it as a FileNotFound expected exception
+    error_handler = new ParseXMLErrorHandler(ParseXMLErrorHandler::Kind::FileNotFound);
+    parser->setErrorHandler(error_handler);
+
+    // Read XML file and obtain the doc object
+    try
+    {
+        // Parsing initially would load the elements in the parser's doc object
+        parser->parse(xercesc::XMLString::transcode(xml_file.c_str()));
+        // Doc would be needed in some other process, so it would be useful to take it apart of the parser
+        doc = parser->getDocument();
+    }
+    // Given file does not exist, so new doc with root element needs to be created
+    catch (const eprosima::qosprof::FileNotFound& ex)
+    {
+        // Implementation would create an empty document
+        doc = implementation->createDocument(0, xercesc::XMLString::transcode(
+            eprosima::qosprof::tag::ROOT), 0);
+        // The root element would be <dds>, with the 'xmlns' att.
+        doc->getDocumentElement()->setAttribute(
+            xercesc::XMLString::transcode(eprosima::qosprof::tag::XMLNS),
+            xercesc::XMLString::transcode(eprosima::qosprof::tag::EPROSIMA_URL));
+    }
+
+    // Initialization process successful
+    return true;
+}
+
+void validate_and_save_xml_document(
+    xercesc::DOMDocument*& doc,
+    ParseXMLErrorHandler*& error_handler,
+    xercesc::DOMLSOutput*& output,
+    xercesc::XercesDOMParser*& parser,
+    xercesc::DOMLSSerializer*& serializer,
+    xercesc::XMLFormatTarget*& target)
+{
+    if (validate_xml(doc, error_handler, output, parser, serializer))
+    {
+        save_xml(doc, output, serializer, target);
+    }
+}
+
+bool validate_xml(
```

Comment:
This is only validating that Xerces can write the value but not validating against the XSD, is not it?